Assets/Scripts/Enemy/Minion/MinionController.cs

```diff
@@ -96,19 +112,31 @@ private IEnumerator moveToAnchor(){
         }
     }
 
-    private void handInResignation(){
+    private void handInResignation()
+    {
         //Self-destruct method that communicates death to the spawner...
-        
+
     }
-}
 
-public enum AttackPattern{
-    Fixed,
-    FixedAlternating,
-    Rotating
-}
+    public void TakeDamage(int damage)
+    {
+        currentHealth -= damage;
+        healthBar.SetHealth((int)(currentHealth/maxhealth*100f));
+        if (currentHealth <= 0)
+        {
+            Destroy(this.gameObject);
+        }
+    }
+    public enum AttackPattern
+    {
+        Fixed,
+        FixedAlternating,
+        Rotating
+    }
 
-public enum AttackRotation{
-    Clockwise,
-    CounterClockwise
-}
+    public enum AttackRotation
```

Comment:
I don't believe this is actually being used anywhere, is it?
Let's remove it so it doesn't confuse anyone (eg, game designers trying to tweak this from the Unity editor)